### PR TITLE
Documenting the class and constructors for the the visualizations root

### PR DESCRIPTION
**Pull request by TorreyBetts**

### CODE REVIEW

File: src/Reveal.Sdk.Dom/Visualizations/OHLCVisualization.cs [hunk 1]
```diff
@@ -3,10 +3,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The OHLC visualization is used for the technical analysis of the price ranges. Like <see cref="CandleStickVisualization"/>, OHLC charts are meant to show the opening, high, low and closing prices for any financial data.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/OHLCVisualization.cs:7]: no need to mention the Candlestick visualization.

File: src/Reveal.Sdk.Dom/Visualizations/RadialVisualization.cs [hunk 2]
```diff
@@ -3,10 +3,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The radial visualization is used for taking data and rendering it as collection of data points wrapped around a circle (rather than stretching along a horizontal line).
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/RadialVisualization.cs:7]: as a collection of data points.... missing the "a". can also drop the parenthesis

File: src/Reveal.Sdk.Dom/Visualizations/TextBoxVisualization.cs [hunk 3]
```diff
@@ -3,9 +3,20 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The text box visualizations is used to display a text body and an optional title. They are not connected to a data source.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/TextBoxVisualization.cs:7]: It is not connected..... Replace "They are" with "It is"

File: src/Reveal.Sdk.Dom/Visualizations/TextViewVisualization.cs [hunk 4]
```diff
@@ -6,10 +6,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The text view visualization is used to display information that follows a key-value pattern. It does so by displaying only the first row of data paired with the column's label.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/TextViewVisualization.cs:10]: Replace "It does so by displaying only" with "It only displays the first..."

File: src/Reveal.Sdk.Dom/Visualizations/AreaChartVisualization.cs [hunk 5]
```diff
@@ -3,12 +3,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The area chart visualization renders as a collection of points connected by straight line segments with the area below the line filled in. Values are represented on the y-axis (labels on the left side) and categories are displayed on the x-axis (bottom labels). This chart emphasize the amount of change over a period of time or compare multiple items as well as the relationship of parts of a whole by displaying the total of the plotted values. Therefore, they are often chronological, showing a change of quantity e.g. accumulation of a commodity over time.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/AreaChartVisualization.cs:7]: remove anything within the parenthesis. At this point people know what an x and y axis are and where they exist on a chart.

File: src/Reveal.Sdk.Dom/Visualizations/BubbleVisualization.cs [hunk 6]
```diff
@@ -6,10 +6,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The bubble chart visualization is a type of scatter chart that show markers with variable scaling to represent the relationship among items in several distinct series of data or to plot data items using x and y coordinates. These coordinates of the data point are determined by two numeric data columns. The bubble chart draws attention to uneven intervals or clusters of data. This chart is often used to plot scientific data, and can highlight the deviation of collected data from predicted results. The bubble chart has many of the characteristics of the scatter chart but with the option to have various radius scale sizes.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/BubbleVisualization.cs:10]: drop the " The bubble chart has many of the characteristics of the scatter chart but with the option to have various radius scale sizes." sentence

File: src/Reveal.Sdk.Dom/Visualizations/CustomVisualization.cs [hunk 7]
```diff
@@ -6,10 +6,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The custom visualizion is used to support visualizations not supported out of the box. This type of visualization uses a web view to pass data to the visualization for display.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/CustomVisualization.cs:10]: Drop "This type of visualization uses a web view to pass data to the visualization for display." This sentence is not applicable within the DOM

File: src/Reveal.Sdk.Dom/Visualizations/DoughnutChartVisualization.cs [hunk 8]
```diff
@@ -3,10 +3,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The doughnut visualization is similar to a pie chart, proportionally illustrating the occurrences of a variable.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/DoughnutChartVisualization.cs:7]: no need to compare to the pie chart. Just tell them what the doughnut chart does

File: src/Reveal.Sdk.Dom/Visualizations/GridVisualization.cs [hunk 9]
```diff
@@ -6,10 +6,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The grid view visualization is used for displaying data which presents the information in a matrix.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/GridVisualization.cs:10]: in a matrix? I think a better term would be a "table"

File: src/Reveal.Sdk.Dom/Visualizations/ImageVisualization.cs [hunk 10]
```diff
@@ -5,10 +5,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The image visualization is used for displaying images, PDFs, or web content.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/ImageVisualization.cs:9]: is PDFs correct? I haven't confirm this is supported

File: src/Reveal.Sdk.Dom/Visualizations/KpiTargetVisualization.cs [hunk 11]
```diff
@@ -6,10 +6,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The KPI (key performance indicator) vs target visualization is used for measuring performance against a target through a prior time period.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/KpiTargetVisualization.cs:10]: you can remove the content within the parenthesis. Customers know what a KPI stands for at this point

File: src/Reveal.Sdk.Dom/Visualizations/KpiTimeVisualization.cs [hunk 12]
```diff
@@ -6,10 +6,24 @@
 
 namespace Reveal.Sdk.Dom.Visualizations
 {
+    /// <summary>
+    /// The KPI (key performance indicator) vs time visualization is used for measuring performance of a value field compared against a prior time period.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/KpiTimeVisualization.cs:10]: you can remove the content within the parenthesis

File: src/Reveal.Sdk.Dom/Visualizations/StepLineChartVisualization.cs [hunk 13]
```diff
@@ -5,10 +5,22 @@ namespace Reveal.Sdk.Dom.Visualizations
 {
     public sealed class StepLineChartVisualization : CategoryVisualizationBase<StepLineChartVisualizationSettings>
     {
+        /// <summary>
+        /// The step line visualization is used for rendering a collection of points connected by continuous vertical and horizontal lines.
```
Comment (brianlagunas) [inline src/Reveal.Sdk.Dom/Visualizations/StepLineChartVisualization.cs:9]: this needs to be on the class

**Review by brianlagunas — CHANGES_REQUESTED** [review]

**Comment by TorreyBetts** [comment]
@brianlagunas for what you reviewed and commented in this PR, should any of these updates be passed on to the technical writers to update our product documentation in cases where the exact wording was used?

**Comment by brianlagunas** [comment]
No. These are API docs and are not be the same as the product documentation.  Usually, I wouldn't write these types of comments because API docs don't need to explain any theory behind the class, just what the class does and how it works.

File: src/Reveal.Sdk.Dom/Visualizations/ImageVisualization.cs (hunk 10 above)
Comment (TorreyBetts) [reply src/Reveal.Sdk.Dom/Visualizations/ImageVisualization.cs:9]: Yes, this is the visualization being used to display PDFs

**Review by brianlagunas — APPROVED** [review]

**Pull request merged**